lib/extensions/rack_session_dalli_patch.rb

```diff
@@ -4,23 +4,35 @@
 module RackSessionDalliPatch
   def delete_sessions(session_ids)
     session_ids.each do |session_id|
-      delete_session(ManageIQ::Session.fake_request, session_id, :drop => true)
+      destroy_session(ManageIQ::Session.fake_request.env, session_id, :drop => true)
```

Comment:
>  delete or destroy_session expects env hash with rack.multithread
> 
> It's unclear how revoke worked before, because even with_lock expected env hash
> and that env hash should have 'rack.multithread' true for our fake request
> passed to revoke since we're using puma with multiple threads.
> 
> Even before with_lock became with_block, it seemed to require this:
> petergoldstein/dalli@9f9c508

@jrafanie the reason that this "worked before" is `delete_session` took the `request` and requested the `.env` from that:

https://github.com/petergoldstein/dalli/commit/9f9c508afab263a2451f2209c4396daf98d33a1b#diff-4b41e4b33e4b095e350045d4e727d2e889890cfc0610d434459f30a267423065L32-L34

So since the method it self just needed something with an object that responded to `env`, that is all that was required.  It just didn't `@mutex` lock when it did work.

It looks like the commit you referenced ( petergoldstein/dalli@9f9c508 ) removed any usage of `env`, but was added back a commit later (same PR):

https://github.com/petergoldstein/dalli/pull/598/commits/819d397fe94c476a9bf875643686ca5dfcad6f14

But that is where `with_block` was then modified to have a different call signature, hence the breakage you ended up seeing.

So yeah, your change looks good, but it did work with the previous implementation, just not with the current versions of `dalli`.

Reply:
Thank you so much for the clarification.  I see it now.  😅